src-tauri/src/mod_management/mod.rs

```diff
@@ -31,3 +35,96 @@ pub fn set_mod_enabled_status(
 
     Ok(())
 }
+
+
+/// Parses `mod.json` for mod name
+// TODO: Maybe pass PathBuf or serde json object
+fn parse_mod_json_for_mod_name(mod_json_path: String) -> Result<String, anyhow::Error> {
+    // Read file into string and parse it
+    let data = std::fs::read_to_string(mod_json_path)?;
+    let parsed_json: serde_json::Value = serde_json::from_str(&data)?;
+
+    // Extract mod name
+    let mod_name = match parsed_json.get("Name").and_then(|value| value.as_str()) {
+        Some(name) => name,
+        None => return Err(anyhow!("No name found")),
+    };
+
+    Ok(mod_name.to_string())
+}
+
+/// Parse `mods` folder for installed mods.
+fn get_installed_mods(game_install: GameInstall) -> Result<Vec<String>, String> {
+    let ns_mods_folder = format!("{}/R2Northstar/mods/", game_install.game_path);
+
+    let paths = std::fs::read_dir(ns_mods_folder).unwrap();
+
+    let mut directories: Vec<PathBuf> = Vec::new();
+    let mut mod_names: Vec<String> = Vec::new();
+
+    // Get list of folders in `mods` directory
+    for path in paths {
+        let my_path = path.unwrap().path();
+
+        let md = std::fs::metadata(my_path.clone()).unwrap();
+        if md.is_dir() {
+            directories.push(my_path);
+        }
+    }
+
+    // Iterate over folders and check if they are Northstar mods
+    for directory in directories {
+        // Check if mod.json exists
+        let mod_json_path = format!("{}/mod.json", directory.to_str().unwrap());
+        if !std::path::Path::new(&mod_json_path).exists() {
+            continue;
+        }
+
+        // Parse mod.json and get mod name
+        let mod_name = match parse_mod_json_for_mod_name(mod_json_path.clone()) {
+            Ok(mod_name) => mod_name,
+            Err(err) => {
+                println!("Failed parsing {} with {}", mod_json_path, err.to_string());
+                continue;
+            }
+        };
+        mod_names.push(mod_name);
+    }
+
+    // Return found mod names
+    Ok(mod_names)
+}
+
+/// Gets list of installed mods and their properties
+/// - name
+/// - is enabled?
+pub fn get_installed_mods_and_properties(
+    game_install: GameInstall,
+) -> Result<Vec<NorthstarMod>, String> {
+    // Get actually installed mods
+    let found_installed_mods = get_installed_mods(game_install.clone())?;
+
+    // Get enabled mods as JSON
+    let enabled_mods: serde_json::Value = match get_enabled_mods(game_install) {
+        Ok(enabled_mods) => enabled_mods,
+        Err(_) => serde_json::from_str("{}").unwrap(), // `enabledmods.json` not found, create empty object
```

Comment:
We should probably create an object with all mods as entries, to solve https://github.com/GeckoEidechse/FlightCore/pull/38#issuecomment-1309080761

Reply:
Eh the object feels kinda overkill, could do but not really necessary and won't solve https://github.com/GeckoEidechse/FlightCore/pull/38#issuecomment-1309080761. That issue is addressed in https://github.com/GeckoEidechse/FlightCore/pull/38#issuecomment-1309493671